libs/agent-sm/agent-policy/src/main/java/org/opensearch/secure_sm/policy/PolicyFile.java

```diff
@@ -166,6 +168,8 @@ private static Permission getKnownPermission(Class<?> claz, String name, String
             return new AllPermission();
         } else if (claz.equals(SecurityPermission.class)) {
             return new SecurityPermission(name, actions);
+        } else if (claz.equals(AttachPermission.class)) {
```

Comment:
Should we consider adding this to `PERM_CLASSES_TO_SKIP`? What''s the advantage of instantiating this if it is a noop?

Reply:
that's also a route. I wanted to keep the list of ```PERM_CLASSES_TO_SKIP``` small and to override any special permissions checks which are rarely used. The  ```AttachPermission``` permissions here now could be used by a new interceptors if ever added to Agent interceptors.